modules/ding_user/ding_user.module

```diff
@@ -682,6 +682,32 @@ function ding_user_default_authname($name) {
 }
 
 /**
+ * Private key variable used to generate tokens.
+ *
+ * Apparently the key generated in drupal_get_private_key() may change,
+ * so we'll get it from 'ding_private_key' variable instead.
+ * Preferably set the 'ding_private_key' variable in local.settings.php
+ *
+ * @return string
+ *   The private key.
+ */
+function ding_get_private_key() {
+  if (!($key = variable_get('ding_private_key', 0))) {
+    if (!($key = variable_get('drupal_private_key', 0))) {
+      $key = drupal_random_key();
+    }
+    watchdog(
+      'ding_user',
+      'New ding_private_key generated. This may break the link to user\'s lists, if on a production site', 
+      array(),
+      WATCHDOG_WARNING
+    );
+    variable_set('ding_private_key', $key);
```

Comment:
I do not agree with this implementation.

As I understand it then by setting a new key if key retrieval fails then we have in essence recreated the problematic behavior from Drupal Core that caused us to implement this change to begin with as discussed with @christianvandel.

I suggest a solution which:

At runtime:
- If `ding_private_key` cannot be retrieved then cause an error - e.g. by throwing an exception. I know that this will cause an error for the end user but the alternative means that we create new user accounts and thus breaking references to lists, comments etc. Since the error should be a temporary situation I think that is preferable.

At install/upgrade time:
- Set `ding_private_key` to `drupal_private_key`.